### PR TITLE
Tweak `Display` impl of `BinomialExtensionField`

### PR DESCRIPTION
**Pull request by dlubarov**
Clippy didn't like the way it was building up a string.

### CODE REVIEW

**Review by LKY-stephen — APPROVED** [review]
LGTM, do we need to update CI?

**Comment by LKY-stephen** [comment]
BTW; This change will show `x^1`.

**Comment by Nashtare** [comment]
Wouldn't we additionally want to skip 0 coeffs? (Even simplify also the case of coeffs being 1)

**Comment by LKY-stephen** [comment]
I quickly write a test for cubicbef, the result seems wrong
`"1656410110 + 1528177553i + 15242231 + 2111298671i*X^2 + 436540726 + 1010126309i*X^3"`

File: field/src/extension/binomial_extension.rs [hunk 1]
```diff
@@ -169,13 +167,11 @@ impl<F: BinomiallyExtendable<D>, const D: usize> Field for BinomialExtensionFiel
...
-        let linear_part = format!("{} + {}*X", self.value[0], self.value[1]); // ok, since D >= 2
-        let nonlin_part: String = self.value[2..]
-            .iter()
-            .zip(2..)
-            .map(|(x, i)| format!(" + {x}*X^{i}"))
-            .collect();
-        write!(f, "{}", linear_part + &nonlin_part)
+        write!(f, "{}", self.value[0])?;
+        for (i, x) in self.value.iter().enumerate().skip(1) {
+            write!(f, " + {}*X^{}", x, i + 1)?;
```
Comment (LKY-stephen) [inline field/src/extension/binomial_extension.rs:172]: should be `i` here; or skip first

**Review by LKY-stephen — CHANGES_REQUESTED** [review]
su

File: field/src/extension/binomial_extension.rs (hunk 1 above)
Comment (dlubarov) [reply field/src/extension/binomial_extension.rs]: Oh right, fixed

**Comment by dlubarov** [comment]
I made the two changes, removing `^1` as the prior code did, and removing zero terms. Also removed multiplication by `1`. Thanks!

File: field/src/extension/binomial_extension.rs [hunk 2]
```diff
@@ -169,13 +170,20 @@ impl<F: BinomiallyExtendable<D>, const D: usize> Field for BinomialExtensionFiel
...
-        let nonlin_part: String = self.value[2..]
+        let str = self
+            .value
             .iter()
-            .zip(2..)
-            .map(|(x, i)| format!(" + {x}*X^{i}"))
-            .collect();
-        write!(f, "{}", linear_part + &nonlin_part)
+            .enumerate()
+            .filter(|(_, x)| !x.is_zero())
```
Comment (LKY-stephen) [inline field/src/extension/binomial_extension.rs:177]: It seems `EF::ZERO` will print nothing,

**Review by LKY-stephen — CHANGES_REQUESTED** [review]

File: field/src/extension/binomial_extension.rs (hunk 2 above)
Comment (dlubarov) [reply field/src/extension/binomial_extension.rs:177]: Good point, fixed and added more tests

**Review by LKY-stephen — APPROVED** [review]

**Pull request merged**